mlx_graphs/data/batch.py

```diff
@@ -0,0 +1,107 @@
+from typing import List, Union
```

Comment:
Importing `List` is not needed anymore - we can directly use `list` (see PEP484 or [here](https://docs.python.org/3/library/typing.html)).